modules/tensor_mechanics/src/materials/ADLAROMANCEStressUpdateBase.C

```diff
@@ -546,39 +732,130 @@ ADLAROMANCEStressUpdateBase::computePolynomial(const ADReal & value,
   }
 }
 
-std::vector<std::vector<std::vector<Real>>>
-ADLAROMANCEStressUpdateBase::getTransformedLimits() const
+std::vector<std::vector<std::vector<std::vector<Real>>>>
+ADLAROMANCEStressUpdateBase::getTransformedLimits(
+    const std::vector<std::vector<std::vector<Real>>> limits)
 {
-  std::vector<std::vector<std::vector<Real>>> transformed_limits(
-      _num_outputs, std::vector<std::vector<Real>>(_num_inputs, std::vector<Real>(2)));
+  std::vector<std::vector<std::vector<std::vector<Real>>>> transformed_limits(
+      _num_tiles,
+      std::vector<std::vector<std::vector<Real>>>(
+          _num_outputs, std::vector<std::vector<Real>>(_num_inputs, std::vector<Real>(3))));
 
-  for (unsigned int i = 0; i < _num_outputs; ++i)
+  for (unsigned int t = 0; t < _num_tiles; ++t)
   {
-    for (unsigned int j = 0; j < _num_inputs; ++j)
+    for (unsigned int i = 0; i < _num_outputs; ++i)
     {
-      for (unsigned int k = 0; k < 2; ++k)
+      transformed_limits[t][i] = limits[t];
```

Comment:
This function needs to be changed to something like this to make it pass the debug test.  ln 748 is changing the size of the size 3 std::vector you created in ln 742.  Then line 754 transformed_limits[t][i][j][2] is out of bounds.  



std::vector<std::vector<std::vector<std::vector<Real>>>>
ADLAROMANCEStressUpdateBase::getTransformedLimits(
    const std::vector<std::vector<std::vector<Real>>> limits)
{
  std::vector<std::vector<std::vector<std::vector<Real>>>> transformed_limits(
      _num_tiles,
      std::vector<std::vector<std::vector<Real>>>(
          _num_outputs, std::vector<std::vector<Real>>(_num_inputs, std::vector<Real>(3))));

  for (unsigned int t = 0; t < _num_tiles; ++t)
  {
    for (unsigned int i = 0; i < _num_outputs; ++i)
    {
      // transformed_limits[t][i] = limits[t];   //limits[t][j].size()=2 and I think that overwrites the size 3
                                                 // you put have on ln 742 so when you try to write to transformed_limits[t][i][j][2] it is out of bounds.
      for (unsigned int j = 0; j < _num_inputs; ++j)
      {
        std::vector<Real> temp_vec(3, 0.0);
        for (unsigned int k = 0; k < 2; ++k)
        {
          temp_vec[k] = limits[t][j][k];
          convertValue(temp_vec[k], _transform[t][i][j], _transform_coefs[t][i][j]);
        }
        temp_vec[2] = (temp_vec[1] - temp_vec[0]) / 2.0;
        transformed_limits[t][i][j] = temp_vec;
      }
    }
  }

  return transformed_limits;
}

Reply:
nice find! How did you figure it out? Did it spit out an error when doing gdb in unix?

Reply:
gdb on unix gave me a backtrace with the line number where you access transformed_limits[t][i][j][2]
maybe valgrind on my mac would have found it.